pymeasure/instruments/attocube/anc300.py

```diff
@@ -300,24 +300,42 @@ def stop_all(self):
             if isinstance(attribute, Axis):
                 attribute.stop()
 
-    def handle_deprecated_host_arg(self, kwargs):
+    def handle_deprecated_host_arg(self, adapter, kwargs):
         """
-        This pulls the host argument from the kwargs and creates a resource string
-        for the VISAAdapter. This is deprecated and separated out to make it easier
-        to remove in the future. This function should be removed and the adapter
-        argument should be made non-optional in the definition of the __init__ function.
+        This function formats user input to the __init__ function to be compatible with the
+        current definition of the __init__ function. This is used to support outdated (deprecated)
+        code. and separated out to make it easier to remove in the future. To whoever removes this:
+        This function should be removed and the `adapter` argument in the __init__ method should
+        be made non-optional.
 
         :param dict kwargs: keyword arguments passed to the __init__ function,
             including the deprecated `host` argument.
         :return str: resource string for the VISAAdapter
         """
-        host = kwargs.pop("host")
-        if not host:
+        host = kwargs.pop("host", None)
+        if not (host or adapter):
+            raise TypeError("ANC300Controller: missing 'adapter' argument")
+
+        if not adapter:
             # because the host argument is deprecated, prompt for the desired
             # argument which is the adapter argument.
-            raise TypeError("ANC300Controller: missing 'adapter' argument")
-        warn("The 'host' argument is deprecated. Use 'adapter' instead.", FutureWarning)
-        return f"TCPIP0::{host}::7230::SOCKET"
+            warn("The 'host' argument is deprecated. Use 'adapter' instead.", FutureWarning)
+            adapter = host
+
+        if isinstance(adapter, str):
+            if adapter.find("::") > -1:
+                # adapter is a resource string, so use it
+                return adapter
+            # otherwise, `adapter` can only be an (deprecated) IPv4 address, so display a
```

Comment:
Hmm, when would the host name contain a "::"?

Reply:
certainly it will not contain `""` , but I would not use "IPv4 address is deprecated" since it could be a hostname as well. I would just rephrase the Warning message and text of the comment

Reply:
Understood. Thanks for the heads up, didn't occur that this was a feature. :)